main/src/library/Fixpoint3/PrecedenceGraph.flix

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// `PrecedenceGraph` contains a graph module which is used to compute a stratification
+/// of a Datalog program. The stratification is based on a topological sort
+/// of the strongly connected components (SCC) of the precedence graph.
+///
+/// The topological sort described a legal stratification of the Datalog program.
+///
+/// A `PseudoSCC` is a group of mutually independent, consecutive SCC in the topological
```

Comment:
Is a pseudo scc also a regular scc? if so, then why not just compute them at initially?